src/handlers/dhcp/DhcpStreamHandler.cpp

```diff
@@ -177,15 +183,43 @@ void DhcpMetricsBucket::process_dhcp_layer(bool deep, pcpp::DhcpLayer *payload,
     case pcpp::DHCP_ACK:
         ++_counters.ACK;
         break;
+    default:
+        break;
     }
 }
+void DhcpMetricsBucket::new_dhcp_transaction(bool deep, pcpp::DhcpLayer *payload, DhcpTransaction &xact)
+{
+    if (!deep) {
+        return;
+    }
+    // lock for write
+    std::unique_lock lock(_mutex);
 
-void DhcpMetricsManager::process_dhcp_layer(pcpp::DhcpLayer *payload, [[maybe_unused]] PacketDirection dir, pcpp::ProtocolType l3, pcpp::ProtocolType l4, [[maybe_unused]] uint32_t flowkey, uint16_t src_port, uint16_t dst_port, timespec stamp)
+    if (auto client_ip = payload->getYourIpAddress(); client_ip.isValid()) {
+        _dhcp_clients.update(client_ip.toString() + "/" + xact.mac_address + "/" + xact.hostname);
```

Comment:
hm do we want/need the ip here? it seems to me mac and hostname actually identify the client. otherwise we will get a new entry if the same mac/host gets a different IP address?

Reply:
I've changed the order of parameters to MAC/HOST/IP

Reply:
done